lib/src/radio-group/radio-group.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, Input, forwardRef, Provider } from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+import { NgClass } from "@angular/common";
+
+export interface RadioGroupItem {
+    /** The label or text to be displayed in the list */
+    label: string;
+    /** optional description to be displayed next to the label */
+    description?: string;
+    /** any value which should be tied to the item */
+    value: any;
+    /** optional disabled flag. Will show the item grayed out and disabled */
+    disabled?: boolean;
+    /** The id or the unique key of the item */
+    key: string;
+}
+
+interface UniqueItem {
+    id: string;
+    optionItem: RadioGroupItem;
+    selected: boolean;
+}
+
+interface DisplayItem extends UniqueItem {
+    className: string;
+}
+
+const CUSTOM_RADIOGROUP_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => RadioGroupComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "sebng-radio-group",
+    templateUrl: "./radio-group.component.html",
+    styleUrls: ["./radio-group.component.scss"],
+    providers: [CUSTOM_RADIOGROUP_CONTROL_VALUE_ACCESSOR],
+})
+export class RadioGroupComponent implements ControlValueAccessor {
+    // TODO: Add support for custom html as well as string labels?
+    @Input("list")
+    set list(value: RadioGroupItem[]) {
+        this._list = value;
+        this._generateHelperArrays();
+    }
+    get list(): RadioGroupItem[] {
+        return this._list;
+    }
+    private _list: RadioGroupItem[];
+
+    @Input() name?: string;
+    @Input() label?: string;
+    @Input("className")
+    set className(value: string) {
+        if (value) {
+            this._classList = ["radio-group", value];
+        }
+    }
+    private _classList: NgClass["ngClass"] = "radio-group";
+
+    @Input() disabled?: boolean = false;
+    @Input() condensed?: boolean = false;
+    @Input() inline?: boolean = false;
+
+    get classList(): NgClass["ngClass"] {
+        return this._classList;
+    }
+
+    // Placeholders for the callbacks which are later provided
+    // by the Control Value Accessor
+    private onTouchedCallback: () => void;
+    private onChangeCallback: (_: any) => void;
+
+    private _selectedValue: RadioGroupItem = null;
+    public allSelected = null;
```

Comment:
missing type

Reply:
actually allSelected should not be there as radio group is not multiple choice, I thought I removed it